pkg/component/watch.go

```diff
@@ -107,10 +107,22 @@ func addRecursiveWatch(watcher *fsnotify.Watcher, path string, ignores []string)
 	return nil
 }
 
+var UserRequestedWatchExit = fmt.Errorf("safely exiting from filesystem watch based on user request")
+
 // WatchAndPush watches path, if something changes in  that path it calls PushLocal
 // ignores .git/* by default
 // inspired by https://github.com/openshift/origin/blob/e785f76194c57bd0e1674c2f2776333e1e0e4e78/pkg/oc/cli/cmd/rsync/rsync.go#L257
-func WatchAndPush(client *occlient.Client, componentName string, applicationName, path string, out io.Writer, ignores []string, delayInterval int) error {
+// Parameters:
+//	client: occlient instance
+//	componentName: Name of component that is to be watched
+//	applicationName: Name of application, the component is part of
+//	path: The path to the source of component(local or binary)
+//	out: io Writer instance
+//	ignores: List/Slice of files/folders in component source, the updates to which need not be pushed to component deployed pod
+//	delayInterval: Interval of time before pushing changes to remote(component) pod
+//	extChan: This is a channel added to terminate the watch command gracefully without passing SIGINT
```

Comment:
please add mention what message should be sent to channel to terminate the watch loop

Reply:
Ack